ccl/multiregionccl: TestRegionLivenessProber failed

ccl/multiregionccl.TestRegionLivenessProber [failed](https://tanzanite.cluster.engflow.com/invocation/536d5b62-1e8e-4866-ab73-4e5a22917fde) on master @ [22d9482e646ae127981386d77b7ebf729346fc3b](https://github.com/cockroachdb/cockroach/commits/22d9482e646ae127981386d77b7ebf729346fc3b):


```
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:87: (TestRegionLivenessProber)
	NOTICE: .Locality() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Locality() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization

pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:87: (TestRegionLivenessProber)
	NOTICE: .Locality() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Locality() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization

pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:87: (TestRegionLivenessProber)
	NOTICE: .Locality() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Locality() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    regionliveness_test.go:149: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:149
        	Error:      	Not equal: 
        	            	expected: 1
        	            	actual  : 2
        	Test:       	TestRegionLivenessProber
        	Messages:   	one region should be dead
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestRegionLivenessProber703254379
--- FAIL: TestRegionLivenessProber (32.51s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionLivenessProber.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34285

ccl/multiregionccl.TestRegionLivenessProber [failed](https://tanzanite.cluster.engflow.com/invocation/ef405f3b-1d27-4975-9d41-0124a9f0de3c) on master @ [8f91e0324af0bd99cbfcc9d44ceeda45ad0960ec](https://github.com/cockroachdb/cockroach/commits/8f91e0324af0bd99cbfcc9d44ceeda45ad0960ec):


```
=== RUN   TestRegionLivenessProber
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestRegionLivenessProber94117358
    test_log_scope.go:81: use -show-logs to present logs inline

pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:87: (TestRegionLivenessProber)
	NOTICE: .Locality() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Locality() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization

pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:87: (TestRegionLivenessProber)
	NOTICE: .Locality() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Locality() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization

pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:87: (TestRegionLivenessProber)
	NOTICE: .Locality() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().Locality() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    regionliveness_test.go:106: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test/pkg/ccl/multiregionccl/regionliveness_test.go:106
        	Error:      	Received unexpected error:
        	            	pq: count-lease timed out reading from a region: operation "count-leases-region" timed out after 514ms (given timeout 500ms): count-leases: failed to read query result: query execution canceled
        	Test:       	TestRegionLivenessProber
    panic.go:541: -- test log scope end --
test logs left over in: outputs.zip/logTestRegionLivenessProber94117358
--- FAIL: TestRegionLivenessProber (160.69s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionLivenessProber.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/multiregionccl/regionliveness_test.go
+++ b/pkg/ccl/multiregionccl/regionliveness_test.go
@@ -46,7 +46,13 @@ import (
 func TestRegionLivenessProber(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	skip.UnderStressRace(t)
+	// This test forces the SQL liveness TTL be a small number,
+	// which makes the heartbeats even more critical. Under stress and
+	// race environments this test becomes even more sensitive, if
+	// we can't send heartbeats within 10 seconds.
+	skip.UnderStress(t)
+	skip.UnderRace(t)
+	skip.UnderDeadlock(t)
 
 	ctx := context.Background()
 
@@ -182,8 +188,13 @@ func TestRegionLivenessProber(t *testing.T) {
 func TestRegionLivenessProberForLeases(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	skip.UnderStressRace(t)
+	// This test forces the SQL liveness TTL be a small number,
+	// which makes the heartbeats even more critical. Under stress and
+	// race environments this test becomes even more sensitive, if
+	// we can't send heartbeats within 10 seconds.
 	skip.UnderStress(t)
+	skip.UnderRace(t)
+	skip.UnderDeadlock(t)
 
 	ctx := context.Background()
 

